dev/tools/bin/generate_gradle_lockfiles.dart

```diff
@@ -0,0 +1,183 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This script generates `android/build.gradle` for each directory specify in the stdin.
+// Then it generate the lockfiles for each Gradle project.
+// To regenerate these files, run `find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart`
+
+import 'dart:io';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:path/path.dart' as path;
+
+void main(List<String> arguments) {
+  print(
+      "Usage: find . -type d -name 'android' | dart dev/tools/bin/generate_gradle_lockfiles.dart\n\n");
+
+  const FileSystem fileSystem = LocalFileSystem();
+  final List<String> androidDirectories = getFilesFromStdin();
+
+  for (final String androidDirectoryPath in androidDirectories) {
+    final Directory androidDirectory =
+        fileSystem.directory(path.normalize(androidDirectoryPath));
+    if (!androidDirectory.existsSync())
+      throw '$androidDirectory does not exist';
+
+    final File rootBuildGradle = androidDirectory.childFile('build.gradle');
+    if (!rootBuildGradle.existsSync()) {
+      print('${rootBuildGradle.path} does not exist - skipping');
+      continue;
+    }
+
+    final File settingsGradle = androidDirectory.childFile('settings.gradle');
+    if (!settingsGradle.existsSync()) {
+      print('${settingsGradle.path} does not exist - skipping');
+      continue;
+    }
+
+    if (!androidDirectory.childDirectory('app').existsSync()) {
+      print('${rootBuildGradle.path} is not an app - skipping');
+      continue;
+    }
+
+    if (!androidDirectory.parent.childFile('pubspec.yaml').existsSync()) {
+      print(
+          '${rootBuildGradle.path} no pubspec.yaml in parent directory - skipping');
+      continue;
+    }
+
+    if (!androidDirectory.parent
+        .childDirectory('lib')
+        .childFile('main.dart')
+        .existsSync()) {
+      print('${rootBuildGradle.path} no main.dart under lib - skipping');
+      continue;
+    }
+
+    print('Processing ${androidDirectory.path}');
+
+    rootBuildGradle.writeAsStringSync(rootGradleFileContent);
+    settingsGradle.writeAsStringSync(settingGradleFile);
+
+    // Fetch pub dependencies.
+    exec('flutter', <String>['pub', 'get'],
+        workingDirectory: androidDirectory.parent.absolute.path);
```

Comment:
style guide says this should either be one line or all new lines with a trailing comma

Reply:
Done